drivers/hmis/lib/hmis_data_cleanup/fix_enrollment_dates_20240416.rb

```diff
@@ -0,0 +1,93 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+class FixEnrollmentDates20240416
+  # For Enrollments in any project EXCEPT special treatment project, where the Entry Date and Exit Date are the same:
+  # - Add one day to the exit date
+  # For Enrollments in special treatment project, where the Exit Date is before the Entry Date (or they are the same):
+  # - If exit date is in 2020, change it to 2023 but same month and day.
+  # For Bed Night Services (recordType:200) in special treatment project, where the DateProvided is in 2020
+  # - Change DateProvided to 2023 (same month and day). The project was not open in 2020
+
+  attr_accessor :special_treatment_project_id
+  def initialize(special_treatment_project_id:)
+    self.special_treatment_project_id = special_treatment_project_id
+  end
+
+  def e_t = Hmis::Hud::Enrollment.arel_table
+  def ex_t = Hmis::Hud::Exit.arel_table
+  def sv_t  = Hmis::Hud::Service.arel_table
+  def p_t = Hmis::Hud::Project.arel_table
+
+  def perform
+    puts "Fixing dates. Special treatment project: #{special_treatment_project_id}"
+    special_project = Hmis::Hud::Project.find_by(project_id: special_treatment_project_id)
+    raise 'Special project not found' unless special_project
+
+    normal_project_ids = Hmis::Hud::Project.where.not(project_id: special_treatment_project_id).pluck(:project_id)
+    fix_dates_special(special_treatment_project_id)
+    fix_dates(normal_project_ids)
+  end
+
+  def fix_dates_special(project_id)
+    puts "Updating (special): #{project_id}"
+
+    exits_to_update = Hmis::Hud::Exit.
```

Comment:
This is at risk of including Exits from other data sources. Add `hmis` to filter to HMIS data source only
```suggestion
    exits_to_update = Hmis::Hud::Exit.hmis.
```